timer/api/timer.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/parser/duration"
+)
+
+// SchedPolicyType is the type of the event schedule policy
+type SchedPolicyType string
+
+const (
+	// SchedEventInterval indicates to schedule events every fixed interval
+	SchedEventInterval SchedPolicyType = "INTERVAL"
+)
+
+// SchedEventPolicy is an interface to tell the runtime how to schedule a timer's events
+type SchedEventPolicy interface {
+	// NextEventTime returns the time to schedule the next timer event. If the second return value is true,
+	// it means we have a next event schedule after `watermark`. Otherwise, it means there is no more event after `watermark`.
+	NextEventTime(watermark time.Time) (time.Time, bool)
```

Comment:
Could you give an example about when the second return value will be "false"?

Reply:
For example, if one timer only wants to execute once, the second return value can be `false` when the watermark is not zero value. But currently only `INTERVAL` is supported, so it is always true.